### ISSUE TITLE
Flink 1.18.1 and 1.19.0 having build jdk 1.8 and causing incompatibilities with Java 17

### ISSUE THREAD

**Issue by Rajat-Pratap** [apache/flink-cdc#3363]
### Please don't check the box below

- [X] I'm aware that bugs and new features should be reported on Apache Jira or Flink mailing list instead of here

### Again, please don't check the box below

- [X] I'm aware that new issues on GitHub will be ignored and automatically closed.

### 请不要勾选以下选项

- [X] 我已知悉缺陷和新功能需要在 Apache Jira 或 Flink 邮件列表中反馈，而不是在这里创建新 issue。

### 也请不要勾选以下选项

- [X] 我已知悉 GitHub 上的新 issue 会被忽略且自动关闭。

**Comment by Rajat-Pratap** [comment]
I am writing flink jobs with latest version for flink (1.18.1). The jobmanager is also deployed with the same version build. But we faced issues when we deployed the jobs. On further investigation, I noticed all libraries from flink have build jdk 1.8. is this the correct version? How to write flink jobs with Java 17 code and which library versions to use? Is Java 17 even supported on the client side or is it just on the server side support for java 17?
